tmail-backend/blob/secondary-blob-store/src/test/java/com/linagora/tmail/blob/secondaryblobstore/SecondaryBlobStoreDAOTest.java

```diff
@@ -0,0 +1,155 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package com.linagora.tmail.blob.secondaryblobstore;
+
+import static org.apache.james.blob.api.BlobStoreDAOFixture.SHORT_BYTEARRAY;
+import static org.apache.james.blob.api.BlobStoreDAOFixture.TEST_BLOB_ID;
+import static org.apache.james.blob.api.BlobStoreDAOFixture.TEST_BUCKET_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.ByteArrayInputStream;
+
+import org.apache.james.blob.api.BlobStoreDAO;
+import org.apache.james.blob.api.BlobStoreDAOContract;
+import org.apache.james.blob.memory.MemoryBlobStoreDAO;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.shaded.com.google.common.collect.ImmutableList;
+
+import com.google.common.io.ByteSource;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class SecondaryBlobStoreDAOTest implements BlobStoreDAOContract {
+    private SecondaryBlobStoreDAO blobStoreDAO;
+    private MemoryBlobStoreDAO firstMemoryBlobStoreDAO;
+    private MemoryBlobStoreDAO secondMemoryBlobStoreDAO;
+
+    @BeforeEach
+    void setUp() {
+        firstMemoryBlobStoreDAO = new MemoryBlobStoreDAO();
+        secondMemoryBlobStoreDAO = new MemoryBlobStoreDAO();
```

Comment:
This is a rare case where I would be happy to see mockito in use, here to inject failures with `spy(...)` and see how the code behaves...

Reply:
for read case, there are already some tests in which read on first storage throw exception (due to data not found), then read on second storage

for write case, partial failure and corresponding tests would be handled in another ticket.